skfem/helpers.py

```diff
@@ -97,3 +97,63 @@ def eye(w, n):
     """Create diagonal matrix with w on diagonal."""
     return np.array([[w if i == j else 0. * w for i in range(n)]
                      for j in range(n)])
+
+
+def det(A):
```

Comment:
Should be easy to make it work also for 2-by-2 matrices? `if A.shape[0]==2: ...`

Reply:
I made these changes. Would test a couple more things. For instance, now that I remember it should be possible to write the inverse entirely using detA and `trace(A^2)` and `trace(A` and implementing these using `np.einsum`. If this is performant then would be valid for matrices of any rank (although I don't know where would these cases arise)

Same goes for `det(A)`.

Reply:
Just tested the Cayley-Hamilton version

```python3
def finv(A):
    detA = vdet(A)
    I1 = einsum("ii...", A)
    I2 = -0.5 * (einsum("ik...,ki...", A, A) - I1**2)
    Asq = einsum("ik...,kj...->ij...", A, A)
    eye = zeros_like(A)
    eye[0, 0] = 1.
    eye[1, 1] = 1.
    eye[2,2] = 1.
    return 1./detA * (Asq - I1 * A + I2 * eye)
```

It's significantly (`~10x`) slower and more complicated than your version. Still faster than `numpy.linalg.inv` on my machine, which for some reason is extremely slow